common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1478,13 +1478,15 @@ public void onComplete(String streamId) throws IOException {
         if (isTooLate(info, partitionInfo.reduceId)) {
           freeDeferredBufs();
           mergeManager.pushMergeMetrics.lateBlockPushes.mark();
+          updateIgnoredBlockBytes();
           throw new BlockPushNonFatalFailure(
             new BlockPushReturnCode(ReturnCode.TOO_LATE_BLOCK_PUSH.id(), streamId).toByteBuffer(),
             BlockPushNonFatalFailure.getErrorMsg(streamId, ReturnCode.TOO_LATE_BLOCK_PUSH));
         }
         if (isStale(info, partitionInfo.appAttemptShuffleMergeId.shuffleMergeId)) {
           freeDeferredBufs();
           mergeManager.pushMergeMetrics.staleBlockPushes.mark();
+          updateIgnoredBlockBytes();
```

Comment:
This is not needed. There is an exception thrown in the next line which triggers `onFailure`